tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,307 @@
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+	assert.NotContains(t, (*resp).JSONResp, "serviceAccount")
+}
+
+func validateCrtbCreationBySteveWithUserPrincipalNameAsInputAndItIsUsed(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	inputData["userPrincipalName"] = "local://" + newUser.ID
+
+	resp, err := client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+	assert.Equal(t, (*resp).JSONResp["userPrincipalName"], inputData["userPrincipalName"])
+}
+
+func validateCrtbCreationBySteveAndInputNotExistedFieldAndItIsIgnored(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["notExisted"] = "filedNotExisted"
+
+	resp, err := client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+	assert.NotContains(t, (*resp).JSONResp, "notExisted")
+}
+
+func validateErrorsWhileCreatingCrtbBySteveAndErroringWithAlreadyCrtbExisted(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+
+	resp, err := client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+
+	// Validate errors when user input existing crtbName while creating crtb
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, "409 Conflict")
+	require.ErrorContains(t, err, "AlreadyExists")
+}
+
+func validateErrorsWhileCreatingCrtbBySteveWhenUserInputNotExistedRoleTemplateName(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["roleTemplateName"] = "roleTemplateNameNotExisted"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
```

Comment:
why are some strings const but not all?

Reply:
I have same question, some of my comments are base don this. IMO, all of the strings const should be in a const for consistency sake.